lib/wallet/txdb.js

```diff
@@ -79,6 +79,62 @@ class TXDB {
     this.wallet.emit(event, data, details);
   }
 
+  /**
+   * Emit events for each type of covenant.
+   * @private
+   * @param {TXRecord} wtx
+   * @param {Details} details
+   */
+
+  async emitCovenants(tx, details, view, height) {
+    const network = this.wdb.network;
+    if (height < 0)
+      height = null;
+
+    const {outputs} = tx;
+    for (const output of outputs) {
+      const {covenant} = output;
+      if (!covenant.isName())
+        continue;
+
+      const nameHash = covenant.get(0);
+      const ns = await view.getNameState(this, nameHash);
+
+      if (EMPTY.equals(ns.name) || !ns.name) {
```

Comment:
The `view` accumulates state during `connectNames`. If the `name` was added to the `NameState` object, then skip the i/o involved with looking it up. Look it up in the covenant data if its present, otherwise query the wallet database for the name based on the `nameHash`.